### ISSUE TITLE
Change the key reuse from NOT RECOMMENDED to MUST NOT

### ISSUE THREAD

**Issue by sophieschmieg** [lamps-wg/draft-composite-sigs#87]
The current key reuse sections merely does NOT RECOMMEND reusing old key material, citing the domain separators added as a defense against out of context (signature stripping) attacks.

However, a domain separator MUST be chosen from a prefix-free set, and in this case we are trying to domain separate against the classical use, which does not include a domain separator at all, i.e. has the empty string as a domain prefix.

The empty string is a prefix of every string, therefore there is no secure way of domain separating the hybrid.

The only alternative I see to a MUST NOT on key reuse is including a domain separator for the classical use case as well, but this will be a breaking change for old clients, and if you introduce a breaking change, you might as well just use a new key instead.

Therefore, I would like us to step up the language from RECOMMEND to MUST.

**Comment by johngray-dev** [comment]
Agreed.  We had an similar issue posted, so I've changed the language to MUST in regards to key re-use.   See #109.    

The key-reuse section also mentions certificate enrollment case, and that a CA SHOULD do a revocation check on the composite key as well as the component keys (in case someone violated the key-reuse guidance).   I made that SHOULD because I don't think all CA's will have access to the low-level composite signature components, so maybe some just won't be able to check the revocations on individual component keys.   Given that we fresh keys MUST be generated, then making this extra check a MUST may be a little heavy handed.  However, such a check is definitely RECOMMENDED.   

Here is the updated text in the pull #109:

> Within the broader context of PQ / Traditional hybrids, we need to consider new attack surfaces that arise due to the hybrid constructions that did not exist in single-algorithm contexts. One of these is key reuse where the component keys within a hybrid are also used by themselves within a single-algorithm context. For example, it might be tempting for an operator to take an already-deployed RSA key pair and combine it with an ML-DSA key pair to form a hybrid key pair for use in a hybrid algorithm. Within a hybrid signature context this leads to a class of attacks referred to as "stripping attacks" discussed in {{sec-cons-non-separability}} and may also open up risks from further cross-protocol attacks. Despite the weak non-separability property offered by the composite signature combiner, key reuse MUST be avoided to prevent the introduction of EUF-CMA vulnerabilities.
> 
> In addition, there is a further implication to key reuse regarding certificate revocation. Upon receiving a new certificate enrollment request, many certification authorities will check if the requested public key has been previously revoked due to key compromise. Often a CA will perform this check by using the public key hash. Therefore, even if both components of a composite have been previously revoked, the CA may only check the hash of the combined composite key and not find the revocations. Therefore, because the possibilty of key reuse exists even though forbidden in this specification, CAs performing revocation checks on a composite key SHOULD also check both component keys independently to verify that the component keys have not been revoked.

**Comment by danvangeest** [comment]
@sophieschmieg 

> The only alternative I see to a MUST NOT on key reuse is including a domain separator for the classical use case as well, but this will be a breaking change for old clients, and if you introduce a breaking change, you might as well just use a new key instead.
> 
> Therefore, I would like us to step up the language from RECOMMEND to MUST.

I can't fault wanting to use MUST instead of RECOMMEND given the state of things.

However, I saw this comment (https://github.com/lamps-wg/draft-composite-sigs/issues/86#issuecomment-2461833968) on a different issue which purports to be a way to get SNS, which would then allow keys to be reused outside of composites.  Assuming implementations have access to the RSA_signhash/ECDSA_signhash (and not some API which does the digest+sign for them), would that be a viable alternative to a MUST NOT?  I ask because there's already been an attempt in the IETF to reuse keys inside and outside composites: https://datatracker.ietf.org/doc/draft-hu-ipsecme-pqt-hybrid-auth/, so I wouldn't be surprised if there are more attempts.  X9 is also doing work which could use keys in and out of a composite-style signatures.  If people are going to be trying to do it anyways, maybe we can give them a template for doing it properly.
